Sources/Confidence/BufferedPassthroughSubject.swift

```diff
@@ -0,0 +1,24 @@
+import Combine
+
+class BufferedPassthrough<T> {
+    private let subject = PassthroughSubject<T, Never>()
+    private var buffer: [T] = []
+    private var isListening = false
+
+    func send(_ value: T) {
+        if isListening {
+            subject.send(value)
+        } else {
+            buffer.append(value)
+        }
+    }
+
+    func publisher() -> AnyPublisher<T, Never> {
+        isListening = true
+        let bufferedPublisher = buffer.publisher
+        buffer.removeAll()
+        return bufferedPublisher
+            .append(subject)
+            .eraseToAnyPublisher()
+    }
```

Comment:
@vahidlazio should we prevent these two functions to ever run simultaneously? I am concerned that some race conditions might cause multiple events to be sent to `subjects` before it is returned as Publisher

Reply:
would be good idea to make `isListening` atomic 👍